pyomo/contrib/mcpp/mcppInterface.cpp

```diff
@@ -1,380 +1,125 @@
-
-    void *ans = new MC(var1);
-
-    return ans;
-}
-
-void *createConstant(double cons)
-{
-    void *ans = new MC(cons);
-
-    return ans;
-}
-
-MC *mult(MC *var1, MC *var2)
-{
-    MC F = *var1 * *var2;
-
-    MC *ans = new MC(F);
-    return ans;
-}
-
-MC *add(MC *var1, MC *var2)
-{
-    MC F = *var1 + *var2;
-
-    MC *ans = new MC(F);
-    return ans;
-}
-
-MC *power(MC *var1, MC *var2)
-{
-    //Extract int value from var2
-    //Pow() will only take an integer value
-    MC var = *var2;
-    double doub = var.l();
-    int exponent = (int)doub;
-
-    MC F = pow(*var1, exponent);
-
-    MC *ans = new MC(F);
-    return ans;
-}
-
-MC *sqrt(MC *var)
-{
-    MC F = sqrt(*var);
-    MC *ans = new MC(F);
-    return ans;
-}
-
-
-MC *monomial(MC *var1, MC *var2)
-{
-    MC F = *var1 * *var2;
-
-    MC *ans = new MC(F);
-    return ans;
-}
-
-MC *reciprocal(MC *var1, MC *var2)
-{
-    MC F = inv(*var2);
-
-    MC *ans = new MC(F);
-    return ans;
-}
-
-MC *negation(MC *var1)
-{
-    MC F = 0 - *var1;
-
-    MC *ans = new MC(F);
-    return ans;
-}
-
-MC *abs(MC *var1)
-{
-    MC F = fabs(*var1);
-
-    MC *ans = new MC(F);
-    return ans;
-}
-
-MC *trigSin(MC *var1)
-{
-    MC F = sin(*var1);
-
-    MC *ans = new MC(F);
-    return ans;
-}
-
-MC *trigCos(MC *var1)
-{
-    MC F = cos(*var1);
-
-    MC *ans = new MC(F);
-    return ans;
-}
-
-MC *trigTan(MC *var1)
-{
-    MC F = tan(*var1);
-
-    MC *ans = new MC(F);
-    return ans;
-}
-
-MC *atrigSin(MC *var1)
-{
-    MC F = asin(*var1);
-
-    MC *ans = new MC(F);
-    return ans;
-}
-
-MC *atrigCos(MC *var1)
-{
-    MC F = acos(*var1);
-
-    MC *ans = new MC(F);
-    return ans;
-}
-
-MC *atrigTan(MC *var1)
-{
-    MC F = atan(*var1);
-
-    MC *ans = new MC(F);
-    return ans;
-}
-
-MC *NPV(MC *var1)
-{
-    MC F = *var1;
-
-    MC *ans = new MC(F);
-    return ans;
-}
-
-void displayOutput(void *ptr)
-{
-    MC *var  = (MC*) ptr;
-    MC F = *var;
-    std::cout << "F: " << F << std::endl;
-}
-
-MC *exponential(MC *var1)
-{
-    MC F = exp(*var1);
-
-    MC *ans = new MC(F);
-    return ans;
-}
-
-MC *logarithm(MC *var1)
-{
-    MC F = log(*var1);
-
-    MC *ans = new MC(F);
-    return ans;
-}
-
-//Get usable information from MC++
-double lower(MC *expr)
-{
-    MC F = *expr;
-    double Flb = F.l();
-    return Flb;
-}
-
-double upper(MC *expr)
-{
-    MC F = *expr;
-    double Fub = F.u();
-    return Fub;
-}
-
-double concave(MC *expr)
-{
-    MC F = *expr;
-    double Fcc = F.cc();
-    return Fcc;
-}
-
-double convex(MC *expr)
-{
-    MC F = *expr;
-    double Fcv = F.cv();
-    return Fcv;
-}
-
-
-double subcc(MC *expr, int index)
-{
-    MC F = *expr;
-    double Fccsub = F.ccsub(index);
-    return Fccsub;
-}
-
-double subcv(MC *expr, int index)
-{
-    MC F = *expr;
-    double Fcvsub = F.cvsub(index);
-    return Fcvsub;
-}
+// Module-level variables as utilities to pass information back to Python
+std::string lastException;
+const char* lastDisplay;
```

Comment:
To (help) prevent the possibility of a memory leak, it would be better if this were a `std::string` and not a `char*`